### PR TITLE
Lexer code has confusion between strings and chars

### PR DESCRIPTION
**Pull request by simon-walton**
Clang complained about this; at least two of these appear
to be actual bugs.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Can one of the admins verify this patch?

**Comment by birarda** [comment]
add to whitelist

**Comment by marko8904** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/7073/app_12369_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/7073/HighFidelity-Beta-PR12369-7dacc2f2c0ead489464aabab268c7563c21cf1a0.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/7073/HighFidelity-Beta-PR12369-7dacc2f2c0ead489464aabab268c7563c21cf1a0.exe

**Comment by AndrewMeadows** [comment]
@birarda, if these were real bugs in the baker... would we be able to contrive some text to parse that would tickle the old bugs?  If so we could make a test-plan: make the baker parse this text and see it break on hifi/master but not in this PR.  If not, what should the test-plan look like for this PR?

**Comment by birarda** [comment]
@AndrewMeadows the JS baking is fully disabled right now. There's a test in the test directory (that must not handle these cases).

I'm okay saying we pass this as changes that don't affect functionality and don't need QA and then re-test things when javascript baking is re-enabled.

**Comment by RyanDowne** [comment]
Is this QA ready? Does it need to go into 66?

File: libraries/baking/src/JSBaker.cpp [hunk 1]
```diff
@@ -190,7 +190,7 @@ bool JSBaker::handleMultiLineComments(QTextStream& in) {
     while (!in.atEnd()) {
         in >> character;
         if (character == '*') {
-            if (in.read(1) == '/') {
+            if (in.read(1)[0] == '/') {
```
Comment (huffman) [inline libraries/baking/src/JSBaker.cpp]: Possible that we are at the end of the stream and there isn't a character to read?

Comment (SimonWalton-HiFi) [reply libraries/baking/src/JSBaker.cpp]: If `in` is a string then won't the original text compare the whole string with "/", not just the next char?
It's unlikely the file wouldn't end in at least a LF; it it's really a problem we could change to
`in.read(1).size() != 0 &&  if (in.read(1)[0] == '/'` .

Comment (birarda) [reply libraries/baking/src/JSBaker.cpp]: Since we're calling `in.read(1)` we'll only ever get back a QString of at most 1 char

Comment (SimonWalton-HiFi) [reply libraries/baking/src/JSBaker.cpp]: OK - we don't this diff then. I can't remember why clang complained now.

Comment (birarda) [reply libraries/baking/src/JSBaker.cpp]: I'm guessing it's because of the QString and char comparison - probably won't complain if we compare to a string

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/8206/app_12369_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/8206/HighFidelity-Beta-PR12369-10f3032746cb4176135b1fd04e5ae91be09f809c.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/8206/HighFidelity-Beta-PR12369-10f3032746cb4176135b1fd04e5ae91be09f809c.exe

**Comment by birarda** [comment]
I added tests to `JSBakerTest.cpp` in #12696 and used those confirm that this is behaving as expected.

**Pull request merged**